tests/bal_business_logic_suite.py

```diff
@@ -216,6 +218,68 @@ def error_cb(idx, batchElementError):
                     self.assertEqual(result.getTraitProperty(trait, property_), value)
 
 
+class Test_resolve_trait_property_expansion(LibraryOverrideTestCase):
+
+    _library = "library_business_logic_suite_var_expansion.json"
+
+    @mock.patch.dict(os.environ, {"CUSTOM": "the value from a custom"})
+    def test_when_envvars_set_then_values_expanded_in_string_properties(self):
+        expected = {
+            "HOME": os.path.expandvars("$HOME"),
+            "CUSTOM": "String with the value from a custom var",
```

Comment:
[Question] I wonder if this is actually testing anything, on windows especially, as `expandvars` will leave a string unchanged if the variable doesn't exists, and I don't think `HOME` is set by default on windows machine, the closest would be `%USERPROFILE%"`. Is it possible we could be checking that `"$HOME" == "$HOME"`?

With the use of the template string substitution, i'm worried about potential weird seperator issues cropping up down the line.

Reply:
For me, `$HOME` was checking a known, existing var, and that our behaviour is matching `os.path.expandvars` there. I guess it does mean it might not do that on windows, so we're not validating that, so worth adding `$USERPROFILE` too for the consistency check there?

Unless someone is using $ for a separator then I think we should be fine. We just behave like a standard templating engine at the moment.

Reply:
I added a check for windows too (as it was as you said) [here](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/23/commits/c6b299ea6a6d7e02a186adf3c5fe5073cc37ef07)

Reply:
So [this failure makes no sense](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/actions/runs/4016814392/jobs/6900345713#step:4:132).

We're doing:
```
trait_data[prop] = string.Template(value).safe_substitute(all_vars)
```

If I run this locally:

![image](https://user-images.githubusercontent.com/12052093/214886104-97d100a9-d38f-470d-a453-65ac4019ab4b.png)

Reply:
![image](https://user-images.githubusercontent.com/12052093/214888880-7697b68d-4c2d-429e-a868-aeab8423cf07.png)

🙃

Reply:
https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/23/commits/471cab66c995154e369c9b7a74d06f5a8c68c149 🤞